packages/leancode_lint/lib/config.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:analyzer/analysis_rule/rule_context.dart';
+// TODO: use a canonical way to access analysis options (https://github.com/dart-lang/sdk/issues/61755)
+// ignore: implementation_imports
+import 'package:analyzer/src/workspace/pub.dart';
+import 'package:yaml/yaml.dart';
+
+class LeancodeLintConfig {
+  LeancodeLintConfig._(this.configMap);
+
+  factory LeancodeLintConfig.fromRuleContext(RuleContext context) {
```

Comment:
This unfortunately reads only analysis_options from the root of the workspace. It is an important feature (and used at LeanCode) that configs are scoped to some directory.

`package:analyzer/src/analysis_options` handles this. See the `AnalysisOptionsProvider` class. It also handles merging options from upper configs and resolves `include:` directives.